op-challenger/game/fault/trace/cannon/provider_test.go

```diff
@@ -121,7 +127,8 @@ func TestGetStepData(t *testing.T) {
 			OracleValue:  []byte{0xdd},
 			OracleOffset: 10,
 		}
-		preimage, proof, data, err := provider.GetStepData(context.Background(), 7000)
+		pos := types.NewPosition(int(provider.gameDepth), 7000)
```

Comment:
We're repeating this pattern of creating a position for a specific trace index a lot in tests - I wonder if its worth having a helper function for it somewhere.

Reply:
Good callout, will do. I think this is a case where we could benefit from being open to using object oriented test suites like the [stretchr testify `suite`](https://github.com/stretchr/testify) since we can move a lot of this redundant unit testing code onto the suite itself as opposed to having one-off setup and builder methods. Happy to explore migrating this to a suite if we're open to it in a follow up pr :)

Reply:
Not sure if that's useful or not as I'm not familiar with it, but even in Java this would likely just be a helper function somewhere, not a test setup type method.  Either way, starting to use suite is more of a team wide kind of discussion. :)

Reply:
Sure, I'm pretty interested in using suites, so I'll try to write something comprehensive up on this.